xarray/core/combine.py

```diff
@@ -50,11 +50,23 @@ def _ensure_same_types(series, dim):
     if series.dtype == object:
         types = set(series.map(type))
         if len(types) > 1:
+            try:
+                import cftime
+
+                cftimes = any(issubclass(t, cftime.datetime) for t in types)
+            except ImportError:
+                cftimes = False
+
             types = ", ".join(t.__name__ for t in types)
-            raise TypeError(
+
+            error_msg = (
                 f"Cannot combine along dimension '{dim}' with mixed types."
                 f" Found: {types}."
             )
+            if cftimes:
+                error_msg = f"{error_msg} Setting `use_cftime=True` may fix this issue."
```

Comment:
Perhaps add "if importing data directly from a file then setting use_cftime..." to lessen confusion in the case that the user got this error purely from calling `combine_by_coords`?

Reply:
Nice, done